benchmarks/automated/tests/utils/cloudwatch.py

```diff
@@ -0,0 +1,59 @@
+import csv
+import os
+import time
+import re
+import boto3
+import uuid
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import numpy as n
+
+from inspect import signature
+from retrying import retry
+from fabric2 import Connection
+from botocore.config import Config
+from botocore.exceptions import ClientError
+
+from invoke import run, sudo
+from invoke.context import Context
+
+from . import DEFAULT_REGION, IAM_INSTANCE_PROFILE, AMI_ID, LOGGER, S3_BUCKET_BENCHMARK_ARTIFACTS
+
+TMP_DIR = "/home/ubuntu"
+LOCAL_TMP_DIR = "/tmp"
+
+
+class CloudWatchMetricsHandler:
+    def __init__(self, context="DevTest", sub_namespace="TestModel"):
+        self.client = boto3.Session(region_name=DEFAULT_REGION).client("cloudwatch")
+        self.context = context
+        self.namespace = f"TorchServe/{context.title()}/{sub_namespace.title()}"
+
+    def push(self, name, unit, value, metrics_info):
+        # dimensions = [{"Name": "BenchmarkContextTest", "Value":self.context}]
+        dimensions = []
+
+        for key in metrics_info:
+            dimensions.append({"Name": key, "Value": str(metrics_info.get(key))})
+
+        try:
+            response = self.client.put_metric_data(
```

Comment:
I wouldn't say so, all updates are atomic, and any failure in case of this put_metric_data will repeat regardless of retries e.g., permission issues, misconfigured namespace or an incorrect unit/datatype. Retrying will not fix such issues, and I don't believe it to be necessary here.